src/Ads/AdsRecommendationsService.php

```diff
@@ -45,30 +54,31 @@ public function __construct( GoogleAdsClient $client ) {
 	}
 
 	/**
-	 * Retrieves recommendations from the database for the specified type and ID.
+	 * Retrieves recommendations from the database for the specified type and campaign ID.
+	 *
+	 * @param array $args {
+	 *     Optional. Arguments to filter recommendations.
 	 *
-	 * @param string $type Optional. Type of recommendation to retrieve. Currently supports only 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH'.
-	 * @param int    $id   Optional. Recommendation ID to filter by. Default 0.
+	 *     @type array $type        Type of recommendations to retrieve.
+	 *     @type int   $campaign_id Campaign ID to filter recommendations.
+	 * }
 	 * @return array Array of recommendations.
 	 */
-	public function get_recommendations( string $type = '', int $id = 0 ): array {
+	public function get_recommendations( array $args = [] ): array {
 		/** @var AdsRecommendationsQuery $query */
 		$query = $this->container->get( AdsRecommendationsQuery::class );
 
-		if ( '' === $type ) {
-			$type = 'IMPROVE_PERFORMANCE_MAX_AD_STRENGTH';
-		}
+		$type        = isset( $args['type'] ) && is_array( $args['type'] ) ? get_valid_recommendation_types( $args['type'] ) : [];
+		$campaign_id = isset( $args['campaign_id'] ) ? (int) $args['campaign_id'] : 0;
```

Comment:
@mukeshpanchal27 should this be `self::get_valid_recommendation_types`?

```suggestion
		$type        = isset( $args['type'] ) && is_array( $args['type'] ) ? self::get_valid_recommendation_types( $args['type'] ) : [];
		$campaign_id = isset( $args['campaign_id'] ) ? (int) $args['campaign_id'] : 0;
```